crates/subspace-service/src/sync_from_dsn/fast_sync_engine.rs

```diff
@@ -0,0 +1,378 @@
+#![allow(dead_code)]
+// TODO: remove after adding the usage
```

Comment:
This module makes little sense to me before there is any use of it. I understand it was based on Substrate internals, but since it seems to be designed to work under specific framework it is hard to review.

Reply:
Try reading the code of `StateStrategy` first. 

This module is very simple (at least I view it as simple after many hours with it): it accepts the target block data, peers to connect, and runs the cycle: 
- do we need more data?
- get the last received key (from the data trie I guess)
- request more data starting from this key
- parse the response, handle the errors
- append received data to the accumulated state
- import the accumulated state when finished

Reply:
I went from here to `StateStrategy` and then to `StateSync` and see that `StateStrategy::new()` only takes things like justifications in order to construct `StateSync`, but what `StateSync` seems to provide is block import, while we don't want block import, we just want to download the state of a particular block and insert it manually. So instead of calling `StateStrategy::new()` we probably want to call something like (currently not public) `StateStrategy::new_with_provider()` where we can provide a "fake" import logic. This raises way more questions than answers for me without seeing in depth how things are connected and depend on each other.

Having `#[allow(dead_code)]` is not helping either.

Ultimately does it even need to be a struct? I'd expect something more like a function that you call and get a state back.